File: packages/go-test/animal/animal.go
package animal

import (
	"errors"
)

var (
	allowed = map[string]bool{
		"cow":   true,
		"horse": true,
		"cat":   true,
		"dog":   true,
		"bird":  true,
	}
)

// Animal represents a simple animal
type Animal struct {
	name    string
	species string
	age     uint
	alive   bool
}

// NewAnimal creates a new animal instance with specified name and type
func NewAnimal(species string, name string) (*Animal, error) {
	if !allowed[species] {
		return nil, errors.New("invalid species")
	}
	return &Animal{
		name:    name,
		age:     0,
		species: species,
		alive:   true,
	}, nil
}

// SayName returns the animals name
func (a *Animal) SayName() (name string) {
	name = a.name
	return name
}

// SaySpecies returns the animals species
func (a *Animal) SaySpecies() (species string) {
	species = a.species
	return species
}

// IsAlive returns true if the animal is alive
func (a *Animal) IsAlive() (living bool) {
	living = a.alive
	return living
}

// HowOld returns the animals age
func (a *Animal) HowOld() (age uint) {
	age = a.age
	return age
}

// Grow increases the animals age by one
func (a *Animal) Grow() (err error) {
	if a.alive != true {
		return errors.New("animal that is dead cannot grow")
	}
	a.age++
	return nil
}

// Kill kills an animal
func (a *Animal) Kill() (err error) {
	if a.alive != true {
		return errors.New("cannot kill what is already dead")
	}
	a.alive = false
	return nil
}

File: packages/typescript/ts-test/src/Animal.ts
/**
 * Animal represents a simple animal
 */
export class Animal {
    /**
     * The allowed types of animals
     */
    private static allowed = ["cat", "dog", "cow", "bird"]

    /**
     * The type of the animal
     */
    private type: string;

    /**
     * The animal's name
     */
    private name: string;

    /**
     * The animals age
     */
    private age: number;

    /**
     * True if the animal is alive
     */
    private alive: boolean;

    /**
     * Create a new animal
     * 
     * @param type the animal's species
     * @param name the animal's name
     */
    constructor(type: string, name: string) {
        if (Animal.allowed.indexOf(type) === -1) {
            throw new Error("invalid type")
        }
        this.type = type;
        this.age = 0;
        this.name = name;
        this.alive = true;
    }

    /**
     * Returns the animals name
     */
    public sayName(): string {
        return this.name;
    }

    /**
     * Returns the animals type (species)
     */
    public sayType(): string {
        return this.type;
    }

    /**
     * Returns the animals age
     */
    public howOld(): number {
        return this.age;
    }

    /**
     * Returns the animals liveness, true if it is alive
     */
    public isAlive(): boolean {
        return this.alive;
    }

    /**
     * Increases the animals age by one
     */
    public grow(): void {
        if (!this.alive) {
            throw new Error("cannot grow when dead");
        }
        this.age++;
    }

    /**
     * Kills the animal. Prevents future growth
     */
    public kill(): void {
        if (!this.alive) {
            throw new Error("cannot kill what is already dead")
        }
        this.alive = false;
    }
}
